lib/ansible/module_utils/ldap.py

```diff
@@ -0,0 +1,64 @@
+COMMON_SPEC={
+    'bind_dn': dict(),
+    'bind_pw': dict(default='', no_log=True),
+    'dn': dict(required=True),
+    'server_uri': dict(default='ldapi:///'),
+    'start_tls': dict(default=False, type='bool'),
+    'validate_certs': dict(default=True, type='bool'),
+}
+
+
+def merge_dict(x, y):
+    z = x.copy()
+    z.update(y)
+
+    return z
+
+
+try:
+    import ldap
+    import ldap.sasl
+
+    HAS_LDAP = True
+except ImportError:
+    HAS_LDAP = False
+
+
+class LdapGeneric(object):
+    def __init__(self, module):
+        # Shortcuts
+        self.module = module
+        self.bind_dn = self.module.params['bind_dn']
+        self.bind_pw = self.module.params['bind_pw']
+        self.dn = self.module.params['dn']
+        self.server_uri = self.module.params['server_uri']
+        self.start_tls = self.module.params['start_tls']
+        self.verify_cert = self.module.params['validate_certs']
+
+        # Establish connection
+        self.connection = self._connect_to_ldap()
+
+    def _connect_to_ldap(self):
```

Comment:
module_utils code is BSD licensed and modules are GNU GPLv3 licensed. I think we need to get sign-off from everyone who contributed to this code snippet before moving it to module_utils. You can refer this [comment](https://github.com/ansible/ansible/pull/26613#issuecomment-318097595) for process.